File: src/main/java/ru/javawebinar/topjava/repository/MealRepository.java
package ru.javawebinar.topjava.repository;

import ru.javawebinar.topjava.model.Meal;

import java.util.List;

public interface MealRepository {

    Meal get(int id);

    Meal save(Meal meal);

    void delete(int id);

    List<Meal> getAll();
}

File: src/main/java/ru/javawebinar/topjava/repository/MealRepository.kt
package ru.javawebinar.topjava.repository

import ru.javawebinar.topjava.model.Meal
import java.time.LocalDate
import java.time.LocalDateTime

interface MealRepository {
    // null if not found, when updated
    fun save(userId: Int, meal: Meal): Meal?

    // false if not found
    fun delete(userId: Int, mealId: Int): Boolean

    // null if not found
    fun get(userId: Int, mealId: Int): Meal?

    fun getAll(userId: Int): List<Meal>

    fun getBetween(userId: Int, start: LocalDateTime, end: LocalDateTime): List<Meal>
}